service/apis/qsl_service.go

```diff
@@ -149,37 +148,47 @@ func CreateFiltersQuery(filterlist string) (string, string, string, error) {
 			// should be 4 elements in matches
 			// the whole string, key, operator, value
 			if len(matches) < 4 {
-				return "", "", "", errors.New("Invalid filters in " + filterlist)
+				return "", "", errors.New("Invalid filters in " + filterlist)
 			}
 
 			keyname := matches[1]
 			operator := matches[2]
 			value := matches[3]
 
-			// if the value is meant to be an int, it won't have quotes around it
-			dectype := ": string"
-			if string(value[0]) != "\"" {
-				dectype = ": int"
+			// json field query
+			// use regex search to match json key and value
+			if strings.Contains(keyname, ".$") {
+				tmp := strings.Split(keyname, ".$")
+				keyname = tmp[0]
+				value = "/\"" + tmp[1] + "\" *: *" + value + "/"
+				if operator != "=" && operator != "~=" {
+					return "", "", errors.New("Filter on json type can only use equal or regexp operator " + filterlist)
+				}
+				operator = "~="
+			} else {
+				// if regrex search, add / and remove " for dgraph query
+				if operator == "~=" {
+					value = "/" + strings.Replace(value, `"`, "", -1) + "/"
+				}
 			}
 
 			// if the value is a string make sure it has quotes on both sides
 			if string(value[0]) == "\"" && !(string(value[len(value)-1]) == "\"") {
-				return "", "", "", errors.New("Invalid filters in " + filterlist)
+				return "", "", errors.New("Invalid filters in " + filterlist)
 			}
 
 			// if keyname is count filter, add prefix cnt_ to the var name
 			if strings.Contains(keyname, "count(") {
 				keyname = "val(cnt_" + keyname[6:]
 			}
 
-			filterdeclaration = filterdeclaration + ", $" + keyname + dectype
 			interfilterfunc = append(interfilterfunc, " "+operatorMap[operator]+"("+keyname+","+value+") ")
 		}
 
 		filterfunc = append(filterfunc, strings.Join(interfilterfunc, "and"))
 
 	}
-	return filterdeclaration, "@filter(" + strings.Join(filterfunc, "or") + ")", paginate, nil
```

Comment:
nice clean-up since the `filterdeclaration` portion of return value wasn't being used anywhere anyway